Set up documentation. Fixes Issue #6

ojeda-e opened the pull request:
This PR fixes issue #6 (cleaner than in PR #8 )

Changes:
Added to `getting_started.rst` 
- [x] Description.

Added to `index.rst`
- [x] Description.
- [x] `Toctree`

Added to `api.rst`
- [x] `Toctree` (although empty for now).

Documentation pages added:
- [x] Algorithm
- [x] Usage (empty for now)
- [x] Visualization

Notes:
- Changes render properly in this PR and `/docs` folder remains `/docs`. 
- Cherry-picked commit [5feb1d4](https://github.com/MDAnalysis/membrane-curvature/pull/13/commits/5feb1d490d20f6591440dcc06b2186e3cdf3ac25) from pending PR #13  that fixes `core.py` 
- Documentation pages added in folder `/source`.

lilyminium commented on docs/conf.py:
```diff
@@ -15,17 +15,18 @@
 # Incase the project was not installed
 import os
 import sys
+sys.path.append(os.path.abspath('../..'))
```
I think this takes you outside your repository (i.e. 1 level up)? Should it be here?

ojeda-e replied:
Yes, that was a mistake. Thanks.

lilyminium replied:
I think `../..` is still 1 level up, as `..` would be the root level directory.

ojeda-e replied:
Ugh, I had deleted the wrong one. Good catch, thanks :)

lilyminium commented on docs/conf.py:
```diff
@@ -15,17 +15,18 @@
 
-project = 'MDAkit MembCurv'
-copyright = ("2021, MDAnalysis. Project structure based on the "
+project = 'MDAkit_membcurv'
```
Just to keep consistent, could you please choose one name between `membrane-curvature` and `MDAkit_membcurv`? My preference is for the former but it's up to you.

ojeda-e replied:
If it's ok, can I instead add an issue fully dedicated to rename `MDAkit MemCurv` to `membrane-curvature`? That change was included in the previous PR and got very messy.

lilyminium replied:
Of course :)

ojeda-e replied:
yay, thanks! ☺️

lilyminium commented on docs/conf.py:
```diff
@@ -79,7 +80,7 @@
 # directories to ignore when looking for source files.
 # This pattern also affects html_static_path and html_extra_path .
-exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store', 'pages/reference-architecture', 'some/other/file.txt']
```
I'm not sure which files these refer to -- are you planning to put these in in the future?

ojeda-e replied:
Removed.

lilyminium commented on docs/index.rst:
```diff
@@ -6,18 +6,37 @@
 Welcome to MDAkit MembCurv's documentation!
 =========================================================
 
+MDAnalysis membrane curvature analysis module calculates gaussian and mean 
```
```suggestion
The MDAnalysis membrane curvature analysis module calculates the Gaussian and mean 
```

I think Gaussian is typically capitalized

ojeda-e replied:
Thanks. Changed.

lilyminium commented on docs/index.rst:
```diff
@@ -6,18 +6,37 @@
+MDAnalysis membrane curvature analysis module calculates gaussian and mean 
+curvature of lipid bilayers from molecular dynamics simulations. 
+
+This module returns a 2-dimensional Numpy array, allowing to visualize values
```
```suggestion
This module returns a 2-dimensional Numpy array, allowing you to visualize values
```

ojeda-e replied:
Changed.

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+Algorithm
+=========================================================
+
+The membrane curvature MDANAlysis module calculates mean and gaussian curvature 
```
```suggestion
The membrane curvature MDAnalysis module calculates the mean and Gaussian curvature 
```

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+from surfaces derived by a group of reference. By default, the group of reference 
+implemented are lipid headgroups.
+
+The algorithm locates all the atoms in the group of reference and map each element 
```
```suggestion
The algorithm locates all the atoms in the group of reference and maps each element 
```

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+determined by the length of the simulation box in the first frame of the 
+trajectory.
+
+The grid is then divided in `n x m` elements, with unit cells of width 2nm (default 
```
```suggestion
The grid is then divided in `n x m` elements, with unit cells of width 2nm (default 
```

I'm not quite following, is `n` and `m` the same as in the previous line? Why is it now transposed? Could you specify that only the x and y coordinates are considered for this mapping?

ojeda-e replied:
No, sorry, it's still `m x n`, not transposed. Change to specified coordinates added.

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+|grid|
+
+Once the grid is populated with the atoms in the group of reference, the respective 
+`z` coordinate is stored in element in an individual array of each `[i, j]`. This 
```
Could you specify that either every z coordinate for every atom in that cell is stored, or the mean coordinate?

ojeda-e replied:
I changed this paragraph a little bit.

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+`z` coordinate is stored in element in an individual array of each `[i, j]`. This 
+operation is performed for every frame in the trajectory. Then, after iterating 
+over every frame of the trajectory, the mean value of the stored `z` coordinate 
+are calculated. These mean values of `z` are the ones used to derive the surface.
```
```suggestion
is calculated. These mean values of `z` are the ones used to derive the surface.
```

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+over every frame of the trajectory, the mean value of the stored `z` coordinate 
+are calculated. These mean values of `z` are the ones used to derive the surface.
+
+With the define surface, then values of mean (`H`) and gaussian (`K`) curvature 
```
```suggestion
With the defined surface, then values of mean (`H`) and Gaussian (`K`) curvature 
```

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,36 @@
+
+With the define surface, then values of mean (`H`) and gaussian (`K`) curvature 
+are calculated. This analysis returns a 2-dimensional array with the values of mean
+mean and gaussian curvature. 
```
```suggestion
mean and gaussian curvature. 
```

Can you specify the size in this array in `m` and `n`?

ojeda-e replied:
Added.

lilyminium replied:
I think this change got lost!

lilyminium reviewed (changes requested):
Thanks for putting this up, @ojeda-e, it gives a good introduction into what happens in the underlying code and how to visualize it! I had a few questions about the content (places where things could be clearer) and notes about the sphinx configuration.

ojeda-e commented:
@lilyminium , I see there is a requested change since the last commit (45031f5). May sound silly but I can't see it. 😅 Could you please show me which change is that?

lilyminium commented:
Hmm, that's weird, I don't think I made one. Maybe you got the notification that I put it in the project tracker?

ojeda-e commented:
This is what I see:

<img width="945" alt="image" src="https://user-images.githubusercontent.com/27988163/121799151-99041e80-cbe7-11eb-9fbf-534758aefa9f.png">

ojeda-e commented:
> Hmm, that's weird, I don't think I made one. Maybe you got the notification that I put it in the project tracker?

Ohhh apparently I had to explicitly request a re-review. :)

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -0,0 +1,47 @@
+
+With the defined surface, then values of mean (`H`) and Gaussian (`K`) curvature
+are calculated. This analysis returns a 2-dimensional array of `m x n` with the values of mean
+mean and Gaussian curvature. The length of this array is `m x n`.
```
Ah ok, this was a source of confusion for me because I wasn't sure if the curvature could be calculated at the edges. That's probably a later discussion though.

lilyminium reviewed (approved):
Thanks for setting this up!
